crypto/crypto.go

```diff
@@ -20,6 +21,27 @@ type (
 		// Sign returns msg's signature and error on failure.
 		Sign(msg []byte) (sig []byte, err error)
 	}
+
+	// Hash is a generic hash interface used by dbft for payloads, blocks and
+	// transactions identification. It is recommended to implement this interface
+	// using hash functions with low hash collision probability. The following
+	// requirements must be provided:
+	// 1. Hashes of two equal payloads/blocks/transactions are equal.
+	// 2. Hashes of two different payloads/blocks/transactions are different.
+	Hash interface {
```

Comment:
> Do you think these belong to crypto?

The original plan was to merge this PR and then move all other crypto-related things to the testing code, so the final content of `crypto` is a set of 4 interfaces (PublicKey, PrivateKey, Hash, Address). We can move Hash and Address to somewhere else, but they need to be placed to a separate package to avoid cyclic dependency problem.

This may be solved after #90, if we move all interfaces to upper level, then there won't be such problem.